src/fiatconnect/saga.ts

```diff
@@ -208,11 +218,66 @@ function* _getFiatAccount({
     providerId,
     fiatConnectProvider.baseUrl
   )
-  const fiatAccount = fiatAccounts.find((account) => account.fiatAccountId === fiatAccountId)
-  if (!fiatAccount) {
-    throw new Error('Could not find fiat account')
+
+  let fiatAccount: FiatAccount | undefined
+  if (fiatAccountType) {
+    fiatAccount = fiatAccounts.find((account) => account.fiatAccountType === fiatAccountType)
+  } else if (fiatAccountId) {
+    fiatAccount = fiatAccounts.find((account) => account.fiatAccountId === fiatAccountId)
+  }
+  return fiatAccount || null
+}
+
+export function* handleSelectFiatConnectQuote({
+  payload: params,
+}: ReturnType<typeof selectFiatConnectQuote>) {
+  const { quote } = params
+  try {
+    // Check for an existing fiatAccount and navigate to Review if we find one
+    // TODO: Also verify that fiatSchemaType matches once it is added to the fiatAccount spec
+    const fiatAccount: FiatAccount = yield call(_getFiatAccount, {
+      fiatConnectProviders: [quote.quote.provider],
+      providerId: quote.getProviderId(),
+      fiatAccountType: quote.getFiatAccountType(),
+    })
+    if (!fiatAccount) {
+      // This is expected when the user has not yet created a fiatAccount with the provider
+      navigate(Screens.FiatDetailsScreen, {
+        quote,
+        flow: quote.flow,
+      })
+      yield delay(500) // to avoid a screen flash
+      yield put(selectFiatConnectQuoteCompleted())
+      return
+    }
+    // Save the fiatAccount in cache
+    yield put(
+      fiatAccountUsed({
+        providerId: quote.getProviderId(),
+        fiatAccountId: fiatAccount.fiatAccountId,
+        fiatAccountType: quote.getFiatAccountType(),
+        flow: quote.flow,
+        cryptoType: quote.getCryptoType(),
+        fiatType: quote.getFiatType(),
+      })
+    )
+    navigate(Screens.FiatConnectReview, {
+      flow: quote.flow,
+      normalizedQuote: quote,
+      fiatAccount,
+    })
+    yield delay(500) // to avoid a screen flash
+    yield put(selectFiatConnectQuoteCompleted())
+  } catch (error) {
+    // Error while attempting fetching the fiatConnect account
+    Logger.debug(
+      TAG,
+      `handleSelectFiatConnectQuote* Error while attempting to fetch the fiatAccount for provider ${quote.getProviderId()}`,
+      error
+    )
+    yield put(selectFiatConnectQuoteCompleted())
```

Comment:
this would just remove the spinner and put the user back on the select provider screen right?

Reply:
yes